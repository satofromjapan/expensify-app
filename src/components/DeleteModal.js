import React from 'react';
import Modal from 'react-modal';

const DeleteModal = ({ deleteRequest, handleRemove, handleCancel }) => (
	<Modal
		isOpen={deleteRequest}
		contentLable="Remove Expense"
		onRequestClose={handleCancel}
		closeTimeoutMS={300}
		className="delete-modal">
		<h3>Remove this expense?</h3>
		<button
			onClick={handleRemove}
			className="button-layout delete-modal__button">
			Remove
		</button>
		<button onClick={handleCancel} className="button-layout button--secondary">
			Cancel
		</button>
	</Modal>
);

export default DeleteModal;
