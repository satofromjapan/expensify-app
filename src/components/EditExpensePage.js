import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { startEditExpense, startRemoveExpense } from '../actions/expenses';

//refactor EditExpensePage to be a class based component
export class EditExpensePage extends React.Component {
  onSubmit = (expense) => {
    this.props.startEditExpense(this.props.expense.id, expense)
    this.props.history.push('/');
  }
  onClick = () => {
    this.props.startRemoveExpense({id: this.props.expense.id})
    this.props.history.push('/');
  }
  render() {
    return (
      <div>
        <ExpenseForm
          expense={this.props.expense}
          onSubmit={this.onSubmit}
        />
        <button onClick={this.onClick}>Remove</button>
      </div>
    )
  }
}
// Set up mapDispatchToProps: editExpense and removeExpense
const mapDispatchToProps = (dispatch, props) => ({
  startEditExpense: (id, expense) => dispatch(startEditExpense(id, expense)),
  startRemoveExpense: (data) => dispatch(startRemoveExpense(data))
})

const mapStatetoProps = (state, props) => {
  return {
    expense: state.expenses.find((expense) => expense.id === props.match.params.id)
  }
};

export default connect(mapStatetoProps, mapDispatchToProps)(EditExpensePage);
