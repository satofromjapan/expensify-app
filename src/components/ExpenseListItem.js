import React from 'react';
import { Link } from 'react-router-dom'
import moment from 'moment'
import numeral from 'numeral'

const ExpenseListItem = ({ id, description, amount, createdAt}) => (
  <div>
    <h3><Link to={`/edit/${id}`}>{description}</Link></h3>
    <p>Amount: {numeral(amount/100).format('$0,0.00')}</p>
    <p>Created At: {moment(createdAt).format('MMMM Do, YYYYY')}</p>
  </div>
)


export default ExpenseListItem;
