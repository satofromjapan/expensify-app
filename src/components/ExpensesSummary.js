import React from 'react'
import { connect } from 'react-redux'
import selectExpensesTotal from '../selectors/expenses-total'
import selectExpenses from '../selectors/expenses'
import numeral from 'numeral'

export const ExpensesSummary = ({expensesCount, expensesTotal}) => (
  <div>
    <h1>{
      expensesCount === 0 ? (
        <p>No expenses to view</p>
      ) : (expensesCount === 1 ? (
        <p>Viewing {expensesCount} expense totalling {numeral(expensesTotal/100).format('$0,0.00')}</p>
      ) : (<p>Viewing {expensesCount} expenses totalling {numeral(expensesTotal/100).format('$0,0.00')}</p>)
      )
    }
    </h1>
  </div>
)

const mapStateToProps = (state) => {
  const visibleExpenses = selectExpenses(state.expenses, state.filters)
  return {
    expensesCount: visibleExpenses.length,
    expensesTotal: selectExpensesTotal(visibleExpenses)
  }
}

export default connect(mapStateToProps)(ExpensesSummary)
