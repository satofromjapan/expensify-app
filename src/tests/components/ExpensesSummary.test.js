import React from 'react'
import { shallow } from 'enzyme'
import {ExpensesSummary} from '../../components/ExpensesSummary'



test('should render ExpensesSummary with one expense', () => {
  const wrapper = shallow(<ExpensesSummary expensesCount={1} expensesTotal={235}/>)
  expect(wrapper).toMatchSnapshot()
})

test('should render ExpensesSummary with multiple expenses', () => {
  const wrapper = shallow(<ExpensesSummary expensesCount={23} expensesTotal={235890798734}/>)
  expect(wrapper).toMatchSnapshot()
})
