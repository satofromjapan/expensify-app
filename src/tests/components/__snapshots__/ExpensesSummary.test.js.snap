// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary with multiple expenses 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExpensesSummary
    expensesCount={23}
    expensesTotal={235890798734}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <h1>
        <p>
                Viewing 
                23
                 expenses totalling 
                $2,358,907,987.34
        </p>
</h1>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <p>
          Viewing 
          23
           expenses totalling 
          $2,358,907,987.34
</p>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Viewing ",
            23,
            " expenses totalling ",
            "$2,358,907,987.34",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Viewing ",
          23,
          " expenses totalling ",
          "$2,358,907,987.34",
        ],
        "type": "p",
      },
      "type": "h1",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h1>
          <p>
                    Viewing 
                    23
                     expenses totalling 
                    $2,358,907,987.34
          </p>
</h1>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <p>
            Viewing 
            23
             expenses totalling 
            $2,358,907,987.34
</p>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Viewing ",
              23,
              " expenses totalling ",
              "$2,358,907,987.34",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Viewing ",
            23,
            " expenses totalling ",
            "$2,358,907,987.34",
          ],
          "type": "p",
        },
        "type": "h1",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`should render ExpensesSummary with one expense 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExpensesSummary
    expensesCount={1}
    expensesTotal={235}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <h1>
        <p>
                Viewing 
                1
                 expense totalling 
                $2.35
        </p>
</h1>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <p>
          Viewing 
          1
           expense totalling 
          $2.35
</p>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Viewing ",
            1,
            " expense totalling ",
            "$2.35",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Viewing ",
          1,
          " expense totalling ",
          "$2.35",
        ],
        "type": "p",
      },
      "type": "h1",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h1>
          <p>
                    Viewing 
                    1
                     expense totalling 
                    $2.35
          </p>
</h1>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <p>
            Viewing 
            1
             expense totalling 
            $2.35
</p>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Viewing ",
              1,
              " expense totalling ",
              "$2.35",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Viewing ",
            1,
            " expense totalling ",
            "$2.35",
          ],
          "type": "p",
        },
        "type": "h1",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
